tests/foreman/sys/test_pulp3_filesystem.py

```diff
@@ -0,0 +1,71 @@
+"""Tests for Pulp3 file system.
+
+:Requirement: Pulp3 installed
+
+:CaseAutomation: Automated
+
+:CaseLevel: System
+
+:CaseComponent: Pulp
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+
+"""
+import pytest
+
+from robottelo.ssh import get_connection
+
+
+@pytest.mark.upgrade
+def test_selinux_status():
+    """Test SELinux satus.
+
+    :id: 43218070-ac5e-4679-b74a-3e2bcb497a0a
+
+    :expectedresults: SELinux is enabled and ther are no denials
+
+    """
+    with get_connection(timeout=100) as connection:
+        # check SELinux is enabled
+        result = connection.run('getenforce')
+        assert 'Enforcing' in result.stdout[0]
+        # check there are no SELinux denials
+        result = connection.run('ausearch -m avc -ts today --raw')
+        assert result.return_code == 0
+
+
+@pytest.mark.upgrade
+def test_file_system_setup():
+    """Test directories and systemd settings for Pulp3 file system layout.
+
+    :id: e584faea-dede-4895-8d7f-411cb074f6c0
+
+    :expectedresults: SELinux is enabled, systemd files are present.
+
+    """
+    with get_connection(timeout=100) as connection:
+        # check pulpcore settings
+        result = connection.run('systemctl cat pulpcore-content.socket')
+        assert result.return_code == 0
+        assert 'ListenStream=/run/pulpcore-content.sock' in result.stdout[3]
+        result = connection.run('systemctl cat pulpcore-content.service')
+        assert result.return_code == 0
+        assert 'Requires=pulpcore-content.socket' in result.stdout[2]
+        result = connection.run('systemctl cat pulpcore-api.service')
+        assert result.return_code == 0
+        assert 'Requires=pulpcore-api.socket' in result.stdout[3]
+        # check pulp3 configuration file present
+        result = connection.run('test -f /etc/pulp/settings.py')
+        assert result.return_code == 0
+        # check pulp3 directories present
+        result = connection.run('test -d /var/lib/pulp')
+        assert result.return_code == 0
+        result = connection.run(
+            '[ -d /var/lib/pulp/assets] && [ -d /var/lib/pulp/pulpcore_static] && \
+            [ -d /var/lib/pulp/media] && [ -d /var/lib/pulp/tmp]'
+        )
+        assert result.return_code == 0
```

Comment:
I think this is hard to debug when it fails. This is probably easier:
```suggestion
        directories = ['/var/lib/pulp', '/...']
        for directory in directories:
            assert connection.run('test -d ' + directory).return_code == 0
```

That gives you a context variable `directory`. I think you can even write:
```python
assert connection.run('test -d ' + directory).return_code == 0, directory + " must exist"
```

Also, looking at this code I'd consider using [pytest-testinfra](https://github.com/pytest-dev/pytest-testinfra) in the future.

Reply:
Hello @ekohl

Thank you for feedback, will test that idea.

BTW, while testing I discovered there is no `/var/lib/pulp/pulpcore_static` on my test system:

`~]# ls /var/lib/pulp/
0005_puppet_module_name_change.txt  content  importers  katello-export  packages   static        tmp     uploads
assets                              exports  imports    media           published  sync_imports  uploas`

Reply:
We may have updated that later on. I think an update of Pulpcore is worked on for 6.9 but not 100% sure.

Reply:
https://github.com/SatelliteQE/robottelo/pull/8682